lego/components.py

```diff
@@ -0,0 +1,144 @@
+"""
+Component object provides the API which tests and libs will use to run code on the component.
+"""
+from __future__ import annotations
+from typing import Optional, Type, TypeVar
+from types import TracebackType
+import abc
+import socket
+import ipaddress
+
+import rpyc
+
+from .connections import BaseConnection, RPyCConnection
+
+T = TypeVar('T', bound='BaseComponent')  # pylint: disable=invalid-name
```

Comment:
Why you use invalid-name?

Reply:
pylint doesn't 'like' one character names, but 'T' is sort of a convention in typing generics.

Reply:
Rename to `Component`